src/annotator.cc

```diff
@@ -69,16 +71,16 @@ void Traverse(const treelite::Tree<ThresholdType, LeafOutputType>& tree,
 template <typename ElementType, typename ThresholdType, typename LeafOutputType>
 inline void ComputeBranchLoopImpl(
     const treelite::ModelImpl<ThresholdType, LeafOutputType>& model,
-    const treelite::DenseDMatrixImpl<ElementType>* dmat, size_t rbegin, size_t rend, int nthread,
-    const size_t* count_row_ptr, uint64_t* counts_tloc) {
-  std::vector<Entry<ElementType>> inst(nthread * dmat->num_col, {-1});
+    const treelite::DenseDMatrixImpl<ElementType>* dmat, size_t rbegin, size_t rend,
+    ThreadConfig thread_config, const size_t* count_row_ptr, uint64_t* counts_tloc) {
```

Comment:
Any reason not to pass `ThreadConfig` by const ref here?

Reply:
Not really. Should this be changed to const ref?

Reply:
I'll go ahead and use const ref here. In a future PR, I may want to add thread scheduling to `ThreadConfig` struct.